migration_sql.go

```diff
@@ -114,11 +114,11 @@ func getSQLStatements(r io.Reader, direction bool) (stmts []string, tx bool) {
 
 	// diagnose likely migration script errors
 	if ignoreSemicolons {
-		log.Println("WARNING: saw '-- +goose StatementBegin' with no matching '-- +goose StatementEnd'")
```

Comment:
Can you please return error and bubble it up to the top-level caller and log.Fatal() in there?

It's a nice opportunity to improve goose error handling.

Reply:
Done, bubbling up the error now instead.

Have added test to handle the parsing cases.

Reply:
I have also added files created by tests to the `.gitignore`

Updated the description @VojtechVitek